### ISSUE TITLE
Using customized configuration file in Python

### ISSUE THREAD

**Issue by sgalal** [BYVoid/OpenCC#406]
目前 Python 版本不能使用自定義的 json 配置。

我新建了一個 `hk2t.json` 如下：

```json
{
  "name": "Traditional Chinese (Hong Kong standard) to Simplified Chinese",
  "segmentation": {
    "type": "mmseg",
    "dict": {
      "type": "ocd2",
      "file": "HKVariantsRevPhrases.ocd2"
    }
  },
  "conversion_chain": [{
    "dict": {
      "type": "group",
      "dicts": [{
        "type": "ocd2",
        "file": "HKVariantsRevPhrases.ocd2"
      }, {
        "type": "ocd2",
        "file": "HKVariantsRev.ocd2"
      }] 
    }
  }]
}
```

問題：

```sh
$ pip install opencc
```

```python
Python 3.9.0a6 (default, May 16 2020, 07:00:10)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from opencc import OpenCC
>>> cc = OpenCC('./hk2t.json')
>>> cc.convert('為了')  # 預期輸出「爲了」
Segmentation fault (core dumped)
```

Command line 可以：

```sh
$ echo '為了' | opencc -c hk2t.json
爲了
```

**Comment by mingruimingrui** [comment]
@sgalal 不好意思，我比较习惯简体字，我们可以用英文吗？

**Comment by mingruimingrui** [comment]
The problem arises because `HKVariantsRevPhrases.ocd2` and `HKVariantsRev.ocd2` cannot be found in the same directory as `hk2t.json`.

**Comment by mingruimingrui** [comment]
But the greater issue is the lack of verbosity.
```python
Segmentation fault (core dumped)
```
tells us nothing

### What causes this behavior?
`cpython` does not recognize c errors and exceptions.
So when the c-library fails, the error message is not propagated to python and python simply returns a generic response.

### Can we solve this issue?
The good news is that I think I've got a working solution now.
https://github.com/mingruimingrui/OpenCC/tree/feature/experimental-pybind

[`pybind11 `](https://github.com/pybind/pybind11) is used to create a python interface for OpenCC.
This way instead of seg-fault, you will get this error message instead.

```python
>>> import opencc
>>> opencc.OpenCC('hk2t.json')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/HOME/miniconda3/envs/opencc/lib/python3.7/site-packages/OpenCC-1.1.0-py3.7-macosx-10.9-x86_64.egg/opencc/__init__.py", line 43, in __init__
    super(OpenCC, self).__init__(config)
ValueError: HKVariantsRevPhrases.ocd2 not found or not accessible.
```

**Comment by sgalal** [comment]
@mingruimingrui Thanks for pointing it out. But I expect the behaviour to be the same as the command line tool.

**Comment by mingruimingrui** [comment]
> @mingruimingrui Thanks for pointing it out. But I expect the behaviour to be the same as the command line tool.

That wont be very convenient to do. The reason command-line works, is because the c library is using PKGDATADIR (derived from CMAKE_INSTALL_PREFIX) to find the config folder.
The goal of the python package is portability, so the same method would not work.

We can add a new feature but I think that your PR is more elegant. https://github.com/BYVoid/OpenCC/pull/408

**Comment by BYVoid** [comment]
Great idea. Thank your very much!

**Comment by sgalal** [comment]
It seems that this issue has been fixed by #414.
